Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -120,34 +125,100 @@ public final class SwiftPackageManagerGraphGenerator: SwiftPackageManagerGraphGe
                                 artifactsFolder: artifactsFolder,
                                 platforms: try condition?.platforms()
                             )
+                        )
+                    case let .product(name, package, condition):
+                        dependencies.append(.thirdPartyTarget(dependency: package, product: name, platforms: try condition?.platforms()))
+                    case let .byName(name, condition):
+                        let platforms = try condition?.platforms()
+                        if packageInfo.targets.contains(where: { $0.name == name }) {
+                            dependencies.append(
+                                Self.localDependency(
+                                    name: name,
+                                    packageInfo: packageInfo,
+                                    artifactsFolder: artifactsFolder,
+                                    platforms: platforms
+                                )
+                            )
                         } else if let package = productToPackage[name] {
-                            return .thirdPartyTarget(dependency: package, product: name, platforms: try condition?.platforms())
+                            dependencies.append(.thirdPartyTarget(dependency: package, product: name, platforms: platforms))
                         } else {
                             throw SwiftPackageManagerGraphGeneratorError.unknownByNameDependency(name)
                         }
                     }
                 }
 
-                return .init(name: target.name, sources: sources, resources: resources, dependencies: dependencies)
+                var cHeaderSearchPaths: [String] = []
+                var cxxHeaderSearchPaths: [String] = []
+                var cDefines: [String: String] = [:]
+                var cxxDefines: [String: String] = [:]
+                var swiftDefines: [String: String] = [:]
+                var cFlags: [String] = []
+                var cxxFlags: [String] = []
+                var swiftFlags: [String] = []
+
+                try target.settings.forEach { setting in
+                    let platforms = try setting.condition?.platforms()
+                    switch (setting.tool, setting.name) {
+                    case (.c, .headerSearchPath):
+                        cHeaderSearchPaths.append(setting.value[0])
+                    case (.c, .define):
+                        let (name, value) = setting.extractDefine
+                        cDefines[name] = value
+                    case (.c, .unsafeFlags):
+                        cFlags.append(contentsOf: setting.value)
+
+                    case (.cxx, .define):
+                        let (name, value) = setting.extractDefine
+                        cxxDefines[name] = value
+                    case (.cxx, .headerSearchPath):
+                        cxxHeaderSearchPaths.append(setting.value[0])
```

Comment:
the fact that `value` is array but sometimes we only care about one value is a little confusing to me. Is this due to the fact that `Setting` struct mirrors what's in SPM? If not, I'd consider saving the value as associated value and the cases would be `SettingName`s

Reply:
Yep, setting is just a 1-1 mirror of the SPM decoded struct, and here is the place where we elaborate it